Pock/Widgets/NowPlaying/NowPlayingHelper.swift

```diff
@@ -74,6 +74,73 @@ class NowPlayingHelper {
         })
     }
     
+    
+    private var artworkAPITask: URLSessionTask?
+    private var artworkDownloadTask: URLSessionTask?
+    
+    var cachedAlbumArtName: String? = nil
+    var cachedAlbumArt: NSImage? = nil
+    
+    func updateArtwork(search: String, completionHandler: @escaping (NSImage?) -> Void) {
+        
+        if (cachedAlbumArtName == search) {
+            completionHandler(cachedAlbumArt)
+            return
+        }
+        
+        
+        // Destroy tasks, if any was already busy
+        if let previousAPITask = artworkAPITask {
+            previousAPITask.cancel()
+        }
+        
+        if let previousDownloadTask = artworkDownloadTask {
+            previousDownloadTask.cancel()
+        }
+        
+        // Start fetching artwork
+        artworkAPITask = URLSession.fetchJSON(fromURL: URL(string: "https://itunes.apple.com/search?term=\(search)&entity=song&limit=1")!) { (data, json, error) in
```

Comment:
I have made it check with the iTunes API only once no matter what because when playing a generic YouTube video for example.
The last used iTunes API image is cached by title and raw data.
Before my last changes the timer loop wasn't working at all but I fixed it. The timer loop is called only 3 times.